pkg/reconcilermanager/controllers/build_names.go

```diff
@@ -32,8 +32,11 @@ func RepoSyncPermissionsName() string {
 	return fmt.Sprintf("%s:%s", configsync.GroupName, core.NsReconcilerPrefix)
 }
 
-// RootSyncPermissionsName returns root reconciler permissions name.
-// e.g. configsync.gke.io:root-reconciler
-func RootSyncPermissionsName() string {
-	return fmt.Sprintf("%s:%s", configsync.GroupName, core.RootReconcilerPrefix)
+// RootSyncPermissionsName returns root reconciler ClusterRoleBinding name.
+// e.g. configsync.gke.io:root-reconciler:my-root-sync
+func RootSyncPermissionsName(reconcilerName string) string {
```

Comment:
I expect there may be a gap here during an upgrade. If the user upgrades from an older version, I expect that the old ClusterRoleBinding (the singleton `configsync.gke.io:root-reconciler`) will still include the Subjects for every `RootSync`. In this case, every pre-existing `RootSync` would still have a `ClusterRoleBinding` which grants `cluster-admin` in addition to the new `ClusterRoleBinding`  (`configsync.gke.io:root-reconciler:my-root-sync`).

Reply:
Yes, I'm aware; see the "Open Question" in the OP. In practice, this will be quite quickly remediated _if_ there is a `RootSync` called `root-sync`, since that one will then "take over" the previously-singleton role binding and remove all the other subjects from it. The hard part to solve is the case when no such `RootSync` exists. Not sure how you prefer to handle that though.

Reply:
Ah I missed that in the PR description and went straight to the design doc, sorry about that.

I'm thinking a potential approach could be:
- remove the special case for `root-reconciler` below, such that its `ClusterRoleBinding` will be named `configsync.gke.io:root-reconciler:root-sync`
- add a snippet to the Reconcile loop which deletes the legacy `ClusterRoleBinding` named `configsync.gke.io:root-reconciler`

This solution should handle the migration in all cases without user interference, even in the case that a `RootSync` named `root-sync` doesn't exist. WDYT?

Reply:
That is cleaner, and makes the name `root-sync` less special. I like it!